src/Model/DataModel/StreamData/HoldEmailData.php

```diff
@@ -137,7 +138,7 @@ public function fixPatronEmail(HoldRequest $holdRequest, Patron $patron): string
         } elseif (count($patron->getEmails()) > 0) {
             return $patron->getEmails()[0];
         } else {
-            throw new APIException(
+            throw new NonRetryableException(
                 'No-email',
                 'Patron did not provide an e-mail address.',
```

Comment:
Message should be the first element here, no? Can probably set 'No-email' as the null in the second param slot. Check the APIException constructor.